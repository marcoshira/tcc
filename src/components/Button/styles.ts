import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};
    color: ${theme.colors.white};
    background: ${theme.colors.primaryColor};
    border-radius: 7px;
    box-shadow: 0px 1px 9px ${theme.colors.primaryColor};
    font-size: ${theme.font.sizes.ltmedium};
    font-weight: 500;
    margin-bottom: 4px;
    transition: ${theme.transitions.faster};

    &:disabled {
      background: ${theme.colors.grey};
      color: ${theme.colors.white};
      border: 2px solid ${theme.colors.grey};
      transition: none;

      &:hover {
        background: ${theme.colors.grey};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.grey};
        cursor: not-allowed;
      }
    }
    &:hover {
      cursor: pointer;
      background: inherit;
      color: ${theme.colors.primaryColor};
    }

    @media ${theme.media.gteMedium} {
      /* width: 33%; */
    }
  `}
`;
