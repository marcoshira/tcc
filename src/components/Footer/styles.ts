import styled, { css } from 'styled-components';

export const Footer = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.primaryColor};
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: white;
    padding: ${theme.spacings.small} 0 ${theme.spacings.medium};

    p {
      margin: 0;
      font-size: 20px;
    }
    a {
      font-size: 18px;
      text-decoration: none;
    }

    @media ${theme.media.gteMedium} {
      padding: ${theme.spacings.small} 8% ${theme.spacings.large};
      flex-direction: row;
      justify-content: space-around;
    }
  `}
`;

export const DevContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    margin-bottom: 20px;

    @media ${theme.media.gteMedium} {
      margin-bottom: 0px;
      p {
        margin-bottom: 20px;
        font-size: 25px;
      }
      a {
        font-size: 25px;
      }
    }
  `}
`;

export const IconsContainer = styled.div`
  ${({ theme }) => css`
    margin-top: 15px;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    p {
      text-align: center;
    }
    svg {
      color: white;
      height: 40px;
    }
  `}
`;

export const IconsInnerContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 20px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    svg {
      color: white;
      height: 40px;
      margin-right: 10px;
    }
    a {
      text-decoration: none;
    }
  `}
`;
