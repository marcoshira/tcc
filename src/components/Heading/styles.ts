import styled, { css } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size ${theme.font.sizes.xlarge}
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, size, uppercase }) => css`
    color: white;
    margin: 0;
    margin-bottom: ${theme.spacings.small};
    padding: ${theme.spacings.large};
    background: ${theme.colors.primaryColor};
    text-align: center;
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
