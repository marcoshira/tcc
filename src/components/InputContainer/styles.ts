import styled, { css } from 'styled-components';

type BorderRed = {
  red?: boolean;
  yellow?: boolean;
};

const redBorder = () => css`
  box-shadow: 0px 0px 5px red;
`;

const yellowBorder = () => css`
  box-shadow: 0px 0px 5px yellow;
`;

export const Wrapper = styled.div<BorderRed>`
  ${({ theme, red, yellow }) => css`
    box-shadow: 0px 1px 9px ${theme.colors.primaryColor};
    border-radius: 7px;
    margin-bottom: ${theme.spacings.small};
    width: 100%;
    transition: ${theme.transitions.slowest};

    ${red && redBorder()}

    @media ${theme.media.gteMedium} {
      margin-bottom: ${theme.spacings.large};
    }

    > div {
      > svg {
        position: absolute;
        top: 12px;
        right: 12px;
        width: 32px;
        color: ${theme.colors.primaryColor};

        &:hover {
          color: ${theme.colors.orange};
        }
      }
      &:hover {
        & .tooltiptext2 {
          visibility: visible;
          opacity: 1;
        }
      }
    }

    & .tooltiptext2 {
      visibility: hidden;
      width: 120px;
      background-color: transparent;
      color: gray;
      text-align: center;
      padding: 15px;
      border-radius: 6px;
      border: 1px solid gray;
      /* height: 100%; */
      width: ${theme.frameSizes.small};
      top: 0px;
      left: 101.5%;

      position: absolute;
      z-index: 1;

      opacity: 0;
      transition: opacity 0.3s;

      text-align: center;
    }
    & .tooltiptext2::before {
      content: ' ';
      position: absolute;
      top: 26px;
      right: 100%;
      margin-top: -10px;
      border-width: 10px;
      border-style: solid;
      border-color: transparent gray transparent transparent;
    }
    & .tooltiptext2::after {
      content: ' ';
      position: absolute;
      top: 26px;
      right: 99.6%;
      margin-top: -10px;
      border-width: 10px;
      border-style: solid;
      border-color: transparent white transparent transparent;
    }

    /* Tooltip container */
    &.tooltip {
      position: relative;
      display: inline-block;
      border-bottom: 1px dotted black; /* If you want dots under the hoverable text */
    }

    /* Tooltip text */
    &.tooltip .tooltiptext {
      visibility: hidden;
      width: 120px;
      background-color: red;
      color: white;
      text-align: center;
      padding: 5px 0;
      border-radius: 6px;
      border: 2px solid red;
      height: ${theme.spacings.large};
      width: 100%;
      bottom: 106%;
      left: 50%;
      margin-left: -50%; /* Use half of the width (120/2 = 60), to center the tooltip */

      /* Position the tooltip text - see examples below! */
      position: absolute;
      z-index: 2;

      opacity: 0;
      transition: opacity 0.3s;
    }

    &.tooltip .tooltiptext4 {
      visibility: hidden;
      width: 120px;
      background-color: #fff3ac;
      color: #856404;
      text-align: center;
      padding: 5px 0;
      border-radius: 6px;
      border: 2px solid #fff3ac;
      height: ${theme.spacings.large};
      width: 100%;
      bottom: 106%;
      left: 50%;
      margin-left: -50%; /* Use half of the width (120/2 = 60), to center the tooltip */

      /* Position the tooltip text - see examples below! */
      position: absolute;
      z-index: 1;

      opacity: 0;
      transition: opacity 0.3s;
    }

    /* Show the tooltip text when you mouse over the tooltip container */
    &.tooltip:hover .tooltiptext,
    &.tooltip:hover .tooltiptext4 {
      visibility: visible;
      opacity: 1;
    }

    &.tooltip .tooltiptext::after {
      content: ' ';
      position: absolute;
      top: 100%; /* At the bottom of the tooltip */
      left: 50%;
      margin-left: -9px;
      border-width: 9px;
      border-style: solid;
      border-color: red transparent transparent transparent;
    }

    &.tooltip .tooltiptext4::after {
      content: ' ';
      position: absolute;
      top: 100%; /* At the bottom of the tooltip */
      left: 50%;
      margin-left: -10px;
      border-width: 10px;
      border-style: solid;
      border-color: #fff3cd transparent transparent transparent;
    }
  `}
`;
