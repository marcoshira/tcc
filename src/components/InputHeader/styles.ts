import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding: 0px 0px;
    display: flex;
    justify-content: center;

    h2 {
      font-size: ${theme.font.sizes.medium};
      font-weight: 300;
    }
  `}
`;
