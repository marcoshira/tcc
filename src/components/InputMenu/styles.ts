import styled, { css } from 'styled-components';
import { InputMenuProps } from '.';

const frameSize = {
  small: (theme) => css`
    height: ${theme.frameSizes.ltxsmall};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.xxsmall};
    }
  `,
  mediumSmall: (theme) => css`
    height: ${theme.frameSizes.small};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.xxsmall};
    }
  `,
  medium: (theme) => css`
    height: ${theme.frameSizes.small};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.ltxsmall};
    }
  `,
  big: (theme) => css`
    height: ${theme.frameSizes.gtsmall};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.small};
    }
  `,
  huge: (theme) => css`
    height: ${theme.frameSizes.smallMedium};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.gtsmall};
    }
  `,

  spt: (theme, depth) => css`
    height: ${depth < 6
      ? theme.frameSizes.xsmall
      : depth < 8
      ? theme.frameSizes.ltsmall
      : depth < 10
      ? theme.frameSizes.small
      : `${(depth - 10) * 2.5 + 26}rem`};
    @media ${theme.media.gteMedium} {
      height: ${`${(Math.floor(depth / 4) + 1) * 6}rem`};
    }
  `,
};

export const Wrapper = styled.div<InputMenuProps>`
  ${({ theme, size, depth }) => css`
    display: flex;
    justify-content: center;
    height: 0px;
    opacity: 0;

    transition: ${theme.transitions.faster};

    > div {
      display: none;
    }

    &.open {
      ${frameSize[size](theme, depth)};
      opacity: 1;

      > div {
        display: flex;
      }
    }

    &.zero {
      height: 0px;
      opacity: 0;
    }

    @media ${theme.media.gteMedium} {
      ${frameSize[size](theme, depth)};
      opacity: 1;

      > div {
        display: flex;
      }
    }
  `}
`;
