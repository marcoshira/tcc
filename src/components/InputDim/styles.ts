import styled, { css } from 'styled-components';
import { Wrapper as NumberInput } from '../NumberInput/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large} 0 ${theme.spacings.medium};
    margin: ${theme.spacings.xsmall} 0 ${theme.spacings.medium};
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
  `}
`;

export const NumberContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;

    ${NumberInput} {
      width: 100%;
      .unit {
        left: 98%;
        top: 10px;
        width: auto;
      }
      span {
        width: 60%;
      }
      input {
        width: 23%;
        height: 23px;
        margin-top: 10px;
      }
    }
    @media ${theme.media.gteMedium} {
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      width: 100%;

      ${NumberInput} {
        width: 50%;
        .unit {
          left: 94%;
          top: 0px;
          width: auto;
        }
        span {
          width: 70%;
          font-size: ${theme.font.sizes.ltmedium};
        }
        input {
          width: 23%;
          height: 23px;
          margin-top: 0px;
        }
      }
    }
  `}
`;

export const RadioContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    width: 100%;

    .control-group {
      display: flex;
      flex-direction: row;
      justify-content: space-around;
      width: 100%;
    }

    .control {
      display: flex;
      justify-content: end;
      position: relative;
      padding-left: 20px;
      margin-bottom: 5px;
      padding-top: 3px;
      cursor: pointer;
      font-size: ${theme.font.sizes.small};
    }
    .control input {
      position: absolute;
      z-index: -1;
      opacity: 0;
    }
    .control_indicator {
      position: absolute;
      top: 8px;
      left: 0px;
      height: 10px;
      width: 10px;
      background: #e6e6e7;
      border: 0px solid #000000;
    }
    .control:hover input ~ .control_indicator,
    .control input:focus ~ .control_indicator {
      background: #cccccc;
    }

    .control input:checked ~ .control_indicator {
      background: #0a1128;
    }
    .control:hover input:not([disabled]):checked ~ .control_indicator,
    .control input:checked:focus ~ .control_indicator {
      background: #0a1128;
    }
    .control input:disabled ~ .control_indicator {
      background: #e6e6e6;
      opacity: 0.6;
      pointer-events: none;
    }
    .control_indicator:after {
      box-sizing: unset;
      content: '';
      position: absolute;
      display: none;
    }
    .control input:checked ~ .control_indicator:after {
      display: block;
    }
    .control-radio .control_indicator {
      border-radius: 50%;
    }

    .control-radio .control_indicator:after {
      left: 7px;
      top: 7px;
      height: 0px;
      width: 0px;
      border-radius: 50%;
      background: #0a1128;
      transition: background 250ms;
    }
    .control-radio input:disabled ~ .control_indicator:after {
      background: #7b7b7b;
    }

    span {
      display: none;
    }

    @media ${theme.media.gteMedium} {
      justify-content: space-evenly;
      width: 80%;
      margin-bottom: ${theme.spacings.small};
      span {
        display: inline;
        font-size: ${theme.font.sizes.ltmedium};
      }

      .control-group {
        width: 60%;
      }

      .control {
        font-size: ${theme.font.sizes.gtsmall};
      }
    }
  `}
`;
