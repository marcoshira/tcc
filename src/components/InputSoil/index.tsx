import { useRef, useState } from 'react';
import * as Styled from './styles';

export type InputSoilProps = {
  onFormChangeSoil?: ({}) => void;
  index?: number;
  reference?: HTMLInputElement;
};

export const InputSoil = ({
  onFormChangeSoil = null,
  reference = null,
  index = 0,
}: InputSoilProps) => {
  const [firstRadio, setFirstRadio] = useState('Areia Comum');
  const [secondRadio, setSecondRadio] = useState('Areia Siltosa');
  const [thirdRadio, setThirdRadio] = useState('Areia Siltoargilosa');
  const [fourthRadio, setFourthRadio] = useState('Areia Argilosa');
  const [fifthRadio, setFifthRadio] = useState('Areia Argilossiltosa');

  const [selected, setSelected] = useState<string>();

  const [type, setType] = useState('Areia');

  const [typeClass1, setTypeClass1] = useState(true);
  const [typeClass2, setTypeClass2] = useState(false);
  const [typeClass3, setTypeClass3] = useState(false);

  const HandleTypeClick = (event) => {
    setType(event.target.id);
    /* istanbul ignore else */
    if (onFormChangeSoil) {
      onFormChangeSoil({ type: event.target.id, subtype: selected });
    }

    switch (event.target.id) {
      case 'Areia':
        setFirstRadio('Areia Comum');
        setSecondRadio('Areia Siltosa');
        setThirdRadio('Areia Siltoargilosa');
        setFourthRadio('Areia Argilosa');
        setFifthRadio('Areia Argilossiltosa');
        setTypeClass1(true);
        setTypeClass2(false);
        setTypeClass3(false);
        break;
      case 'Silte':
        setFirstRadio('Silte Comum');
        setSecondRadio('Silte Arenoso');
        setThirdRadio('Silte Arenoargiloso');
        setFourthRadio('Silte Argiloso');
        setFifthRadio('Silte Argiloarenoso');
        setTypeClass1(false);
        setTypeClass2(true);
        setTypeClass3(false);
        break;
      case 'Argila':
        setFirstRadio('Argila Comum');
        setSecondRadio('Argila Arenosa');
        setThirdRadio('Argila Arenossiltosa');
        setFourthRadio('Argila Siltosa');
        setFifthRadio('Argila Siltoarenosa');
        setTypeClass1(false);
        setTypeClass2(false);
        setTypeClass3(true);
        break;
    }
  };

  const handleChange = (event) => {
    setSelected(event.target.id);
    /* istanbul ignore else */
    if (onFormChangeSoil) {
      onFormChangeSoil({ type: type, subtype: event.target.id });
    }
  };

  return (
    <Styled.Wrapper>
      <Styled.TypeContainer>
        <Styled.Type
          onClick={HandleTypeClick}
          className={typeClass1 && 'selected'}
          id="Areia"
        >
          Areia
        </Styled.Type>
        <Styled.Type
          onClick={HandleTypeClick}
          className={typeClass2 && 'selected'}
          id="Silte"
        >
          Silte
        </Styled.Type>
        <Styled.Type
          onClick={HandleTypeClick}
          className={typeClass3 && 'selected'}
          id="Argila"
        >
          Argila
        </Styled.Type>
      </Styled.TypeContainer>
      <Styled.SubTypeContainer>
        <div className="control-group">
          <label className="control control-radio">
            {firstRadio}
            <input
              type="radio"
              name={`soil ${index}`}
              value={firstRadio}
              id="1"
              onClick={handleChange}
            />
            <div className="control_indicator"></div>
          </label>
          <label className="control control-radio">
            {secondRadio}
            <input
              type="radio"
              name={`soil ${index}`}
              value={secondRadio}
              id="2"
              onClick={handleChange}
            />
            <div className="control_indicator"></div>
          </label>
          <label className="control control-radio">
            {thirdRadio}
            <input
              type="radio"
              name={`soil ${index}`}
              value={thirdRadio}
              id="3"
              onClick={handleChange}
            />
            <div className="control_indicator"></div>
          </label>
          <label className="control control-radio">
            {fourthRadio}
            <input
              type="radio"
              name={`soil ${index}`}
              value={fourthRadio}
              id="4"
              onClick={handleChange}
            />
            <div className="control_indicator"></div>
          </label>
          <label className="control control-radio">
            {fifthRadio}
            <input
              type="radio"
              name={`soil ${index}`}
              value={fifthRadio}
              id="5"
              onClick={handleChange}
            />
            <div className="control_indicator"></div>
          </label>
        </div>
      </Styled.SubTypeContainer>
    </Styled.Wrapper>
  );
};
