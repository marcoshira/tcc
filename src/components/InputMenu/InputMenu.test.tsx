import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { InputMenu, InputMenuProps } from '.';
import { InputSoil } from '../InputSoil';

const props: InputMenuProps = {
  children: <InputSoil />,
};

describe('<InputMenu />', () => {
  it('should render', () => {
    renderTheme(<InputMenu {...props} />);

    expect(screen.getByText('Areia')).toBeInTheDocument();
  });

  it('should render no height and opacity', () => {
    const { container } = renderTheme(<InputMenu {...props} />);

    expect(container.firstChild).toHaveStyle({
      height: '0px',
      opacity: '0',
    });

    expect(container).toMatchSnapshot();
  });

  it('should render the default height and opacity', () => {
    const { container } = renderTheme(<InputMenu {...props} display={true} />);

    expect(container.firstChild).toHaveStyle({
      height: '24rem',
      opacity: '1',
    });
  });

  it('should render small height ', () => {
    const { container } = renderTheme(
      <InputMenu {...props} display={true} size="small" />,
    );

    expect(container.firstChild).toHaveStyle({
      height: '18rem',
    });
  });

  it('should render mediumSmall height ', () => {
    const { container } = renderTheme(
      <InputMenu {...props} display={true} size="mediumSmall" />,
    );

    expect(container.firstChild).toHaveStyle({
      height: '24rem',
    });
  });

  it('should render big height ', () => {
    const { container } = renderTheme(
      <InputMenu {...props} display={true} size="big" />,
    );

    expect(container.firstChild).toHaveStyle({
      height: '28rem',
    });
  });

  it('should render huge height ', () => {
    const { container } = renderTheme(
      <InputMenu {...props} display={true} size="huge" />,
    );

    expect(container.firstChild).toHaveStyle({
      height: '32rem',
    });
  });
  it('should render spt < 6 height ', () => {
    const { container } = renderTheme(
      <InputMenu {...props} display={true} size="spt" depth={2} />,
    );

    expect(container.firstChild).toHaveStyle({
      height: '18rem',
    });
  });
  it('should render spt < 8 height ', () => {
    const { container } = renderTheme(
      <InputMenu {...props} display={true} size="spt" depth={7} />,
    );

    expect(container.firstChild).toHaveStyle({
      height: '21rem',
    });
  });
  it('should render spt < 10 height ', () => {
    const { container } = renderTheme(
      <InputMenu {...props} display={true} size="spt" depth={8} />,
    );

    expect(container.firstChild).toHaveStyle({
      height: '24rem',
    });
  });
  it('should render spt > 10 height ', () => {
    const { container } = renderTheme(
      <InputMenu {...props} display={true} size="spt" depth={14} />,
    );
    expect(container.firstChild).toHaveStyle({
      height: '36rem',
    });
  });
  it('should render 0 layer height ', () => {
    const { container } = renderTheme(
      <InputMenu {...props} display={true} size="layer" layer={0} />,
    );

    expect(container.firstChild).toHaveStyle({
      height: '3rem',
    });
  });
  it('should render layer height ', () => {
    const { container } = renderTheme(
      <InputMenu {...props} display={true} size="layer" layer={5} />,
    );

    expect(container.firstChild).toHaveStyle({
      height: '169rem',
    });
  });
});
