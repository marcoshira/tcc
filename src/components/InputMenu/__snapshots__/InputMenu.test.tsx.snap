// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputMenu /> should render no height and opacity 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 0px;
  opacity: 0;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c0 > div {
  display: none;
}

.c0.open {
  height: 24rem;
  opacity: 1;
}

.c0.open > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0.zero {
  height: 0px;
  opacity: 0;
}

.c1 {
  width: 80%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  color: #0A1128;
  font-size: 2rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c3.selected {
  color: rgb(207,207,45);
}

.c3:hover {
  color: rgb(230,159,28);
  cursor: pointer;
}

.c4 {
  margin: 2.4rem 0;
}

.c4 .control {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
  position: relative;
  padding-left: 20px;
  margin-bottom: 5px;
  padding-top: 3px;
  cursor: pointer;
  font-size: 1.6rem;
}

.c4 .control input {
  position: absolute;
  z-index: -1;
  opacity: 0;
}

.c4 .control_indicator {
  position: absolute;
  top: 8px;
  left: 0px;
  height: 10px;
  width: 10px;
  background: #e6e6e7;
  border: 0px solid #000000;
}

.c4 .control:hover input ~ .control_indicator,
.c4 .control input:focus ~ .control_indicator {
  background: #cccccc;
}

.c4 .control input:checked ~ .control_indicator {
  background: #0a1128;
}

.c4 .control:hover input:not([disabled]):checked ~ .control_indicator,
.c4 .control input:checked:focus ~ .control_indicator {
  background: #0a1128;
}

.c4 .control input:disabled ~ .control_indicator {
  background: #e6e6e6;
  opacity: 0.6;
  pointer-events: none;
}

.c4 .control_indicator:after {
  box-sizing: unset;
  content: '';
  position: absolute;
  display: none;
}

.c4 .control input:checked ~ .control_indicator:after {
  display: block;
}

.c4 .control-radio .control_indicator {
  border-radius: 50%;
}

.c4 .control-radio .control_indicator:after {
  left: 7px;
  top: 7px;
  height: 0px;
  width: 0px;
  border-radius: 50%;
  background: #0a1128;
  -webkit-transition: background 250ms;
  transition: background 250ms;
}

.c4 .control-radio input:disabled ~ .control_indicator:after {
  background: #7b7b7b;
}

@media (min-width:768px) {
  .c0.open {
    height: 15rem;
  }
}

@media (min-width:768px) {
  .c0 {
    height: 24rem;
    opacity: 1;
  }

  .c0 > div {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c1 {
    width: 90%;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

@media (min-width:768px) {
  .c4 .control {
    text-align: center;
  }
}

@media (min-width:768px) {
  .c4 .control-group {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }

  .c4 .control {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: end;
    -webkit-justify-content: end;
    -ms-flex-pack: end;
    justify-content: end;
    position: relative;
    margin-bottom: 5px;
    padding-top: 3px;
    cursor: pointer;
    font-size: 1.6rem;
  }
}

<div>
  <div
    aria-label="InputMenu"
    class="c0 false false"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          class="c3 selected"
          id="Areia"
        >
          Areia
        </a>
        <a
          class="c3"
          id="Silte"
        >
          Silte
        </a>
        <a
          class="c3"
          id="Argila"
        >
          Argila
        </a>
      </div>
      <div
        class="c4"
      >
        <div
          class="control-group"
        >
          <label
            class="control control-radio"
          >
            Areia Comum
            <input
              id="1"
              name="soil 0"
              type="radio"
              value="Areia Comum"
            />
            <div
              class="control_indicator"
            />
          </label>
          <label
            class="control control-radio"
          >
            Areia Siltosa
            <input
              id="2"
              name="soil 0"
              type="radio"
              value="Areia Siltosa"
            />
            <div
              class="control_indicator"
            />
          </label>
          <label
            class="control control-radio"
          >
            Areia Siltoargilosa
            <input
              id="3"
              name="soil 0"
              type="radio"
              value="Areia Siltoargilosa"
            />
            <div
              class="control_indicator"
            />
          </label>
          <label
            class="control control-radio"
          >
            Areia Argilosa
            <input
              id="4"
              name="soil 0"
              type="radio"
              value="Areia Argilosa"
            />
            <div
              class="control_indicator"
            />
          </label>
          <label
            class="control control-radio"
          >
            Areia Argilossiltosa
            <input
              id="5"
              name="soil 0"
              type="radio"
              value="Areia Argilossiltosa"
            />
            <div
              class="control_indicator"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
